.github/workflows/build_and_test.yml

```diff
@@ -319,3 +319,29 @@ jobs:
         ./dev/change-scala-version.sh 2.13
         ./build/mvn $MAVEN_CLI_OPTS -DskipTests -Pyarn -Pmesos -Pkubernetes -Phive -Phive-thriftserver -Phadoop-cloud -Djava.version=11 -Pscala-2.13 install
         rm -rf ~/.m2/repository/org/apache/spark
+
+  scala-213-sbt:
+    name: Scala 2.13 build with SBT
```

Comment:
Hm, should we set this job in the PR builder? The (almost) same thing is tested at "Scala 2.13 build". I think we'll set a Jenkins regular job to run the tests at SPARK-33044 as well. I would avoid adding too many jobs here since we're not able to have more resources (see also SPARK-32264).

Reply:
> The (almost) same thing is tested at "Scala 2.13 build"

The main purpose of adding the build with SBT is not just testing with SBT. As I mentioned, some compilation error happens only with SBT so I'd like detect such error before any changes being merged. 
I understand we have not enough resource for GitHub Action but if an additional build is added to the Jenkins regular job, it will take much more longer time doesn't it?
Are there any problem to replace the existing "Scala 2.13 build" with the new one?

Reply:
I am okay with replacing the existing "Scala 2.13 build". Seems that'll catch the compilation errors more correctly. 

> it will take much more longer time doesn't it?

It does takes longer to detect the errors. However, there are many other combinations to test and currently we can't test all combinations. So I think the default common-ish combination was picked to use it as a GitHub Actions and Jenkins PR builder. For example, JDK 11 is not being tested in GitHub Actions and Jenkins PR builder.

Reply:
> For example, JDK 11 is not being tested in GitHub Actions and Jenkins PR builder.

Yeah, I know but whether build is successfully finish is tested in GitHub Actions so I thought build test (not unit test) should be checked in GitHub Actions too for now.
Anyway, If we can replace the existing "Scala 2.13 build" with the new one, I'll do it.
@dongjoon-hyun Do you have any concern about replacing it?